src/nginx/http.cc

```diff
@@ -1104,20 +1102,20 @@ Status ngx_esp_create_http_request(
     ngx_log_t *log, HTTPRequest *request,
     ngx_esp_http_connection **out_http_connection) {
   // Create the connection pool and request pools.
+  std::unique_ptr<ngx_pool_t, ngx_pool_t_deleter> esp_pool;
   std::unique_ptr<ngx_pool_t, ngx_pool_t_deleter> connection_pool;
   std::unique_ptr<ngx_pool_t, ngx_pool_t_deleter> request_pool;
 
-  Status status = allocate_pools(log, &connection_pool, &request_pool);
+  Status status = allocate_pools(log, esp_pool, connection_pool, request_pool);
```

Comment:
-> esp_egress_pool or esp_outbound_pool?

Reply:
Please also consider adding tests to protect, if there is an easy way.

Reply:
All code in http.cc is for making side-way calls. There are 3 pools,  owned by esp, connection an d request. There variables are called by their owners.

Reply:
For testing, I have manually tested without clearing content-length. if content-length > 4kb, postpone filter will generate an error, nginx will free the connection pool and request pool. 

But I like to keep the code of clearing content-length to avoid postpone filter to log an error which may cause false alarm to customers.